backend/app/websocket/compression.py

```
@@ -0,0 +1,57 @@
+    def __init__(self, compression_level: int = 6, min_size_for_compression: int = 1024, 
+                 min_compression_ratio: float = 0.1):
+        self.compression_level = compression_level
+        self.min_size_for_compression = min_size_for_compression
+        self.min_compression_ratio = min_compression_ratio
+    
+    def compress(self, message: str) -> bytes:
+        """Compress message using zlib"""
+        try:
+            return zlib.compress(message.encode(), self.compression_level)
+        except zlib.error as e:
+            logger.error(f"Compression failed: {e}")
```

Ruff (0.12.2)

Undefined name `logger`

(F821)